src/SFA.DAS.Tools.Support.Web/Controllers/ApprovalsController.cs

```diff
@@ -19,29 +24,43 @@ public class ApprovalsController : Controller
         private readonly ILogger<ApprovalsController> _logger;
         private readonly IEmployerCommitmentsService _employerCommitmentsService;
         private readonly IMapper _mapper;
-        private readonly string emailClaim = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress";
-        private readonly string nameClaim = "name";
-        private readonly string userIdClaim = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier";
+        private readonly IOptions<ClaimsConfiguration> _claimConfiguration;
 
-        public ApprovalsController(ILogger<ApprovalsController> logger, IEmployerCommitmentsService employerCommitmentsService, IMapper mapper)
+        public ApprovalsController(ILogger<ApprovalsController> logger, IEmployerCommitmentsService employerCommitmentsService, IMapper mapper, IOptions<ClaimsConfiguration> claimConfiguration)
         {
             _logger = logger;
             _employerCommitmentsService = employerCommitmentsService;
             _mapper = mapper;
+            _claimConfiguration = claimConfiguration;
+            _claimConfiguration.Value.ValidateConfiguration();
         }
 
         [HttpGet("searchApprenticeships", Name = ApprovalsRouteNames.SearchApprenticeships)]
-        public IActionResult SearchApprenticeships()
+        public IActionResult SearchApprenticeships(string employerName, string courseName, string providerName, string apprenticeName, DateTime? startDate, DateTime? endDate, string selectedStatus)
         {
-            return View(new SearchApprenticeshipsViewModel());
+            var sDate = startDate.HasValue && startDate.Value != DateTime.MinValue ? startDate : null;
+            var eDate = endDate.HasValue && endDate.Value != DateTime.MinValue ? endDate : null;
+            var status = string.IsNullOrWhiteSpace(selectedStatus) ? "0" : selectedStatus;
+
+            var model = new SearchApprenticeshipsViewModel
+            {
+                EmployerName = employerName,
+                CourseName = courseName,
+                ProviderName = providerName,
+                StartDate = sDate,
+                EndDate = eDate,
+                SelectedStatus = status,
+                ApprenticeName = apprenticeName
+            };
+
+            return View(model);
         }
 
         [HttpPost("searchApprenticeships", Name = ApprovalsRouteNames.SearchApprenticeships)]
```

Comment:
I would separate out Actions that return Json into a 2nd "data" controller

Reply:
Will separate out into a second controller.